sources/models/src/modeled_types/mod.rs

```diff
@@ -29,6 +29,18 @@ pub mod error {
         ))]
         InvalidIdentifier { input: String },
 
+        #[snafu(display(
+            "Kernel boot config keywords may only contain ASCII alphanumerics plus hyphens and underscores, received '{}'",
```

Comment:
Nit: there is a similar text a few lines above, I think you could be consistent and use (almost?) the same message.

Reply:
I'd rather keep the error variants separate since they're for different model types that have similar but different format requirements.